src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,322 @@
+      std::vector<double>::iterator max_it = std::max_element(experiment_data_.begin(), experiment_data_.end());
+      uint64_t experiment_data_peak_idx = std::distance(experiment_data_.begin(), max_it);
+      experiment_data_peak = experiment_data_[experiment_data_peak_idx];
+      
+      if (experiment_data_peak_idx >= 1 && experiment_data_peak != 0.0)
+      {
+        double previous_experiment_data_peak = experiment_data[experiment_data_peak_idx - 1];
+        double isotopic_purity = experiment_data_peak_idx / (experiment_data_peak_idx + (previous_experiment_data_peak / experiment_data_peak));
+        featuremap_with_isotopic_purity.setMetaValue(isotopic_purity_name, isotopic_purity);
+      }
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateIsotopicPurities(
+    const FeatureMap& normalized_featureMap,
+    FeatureMap& featureMap_with_isotopic_purity,
+    const std::vector<double>& experiment_data,
+    const std::string& isotopic_purity_name)
+  {
+    for (const Feature& feature : normalized_featureMap)
+    {
+      Feature feature_with_isotopic_purity;
+      calculateIsotopicPurity(feature, feature_with_isotopic_purity, experiment_data, isotopic_purity_name);
+      featureMap_with_isotopic_purity.push_back(feature_with_isotopic_purity);
+    }
+  }
+  
+  void IsotopeLabelingMDVs::calculateMDVAccuracy(
+    const Feature& normalized_feature,
+    Feature& feature_with_accuracy_info,
+    const std::vector<double>& fragment_isotopomer_measured,
+    const std::string& fragment_isotopomer_theoretical_formula)
+  {
+    feature_with_accuracy_info = normalized_feature;
+    
+    std::vector<double> fragment_isotopomer_theoretical;
+    
+    IsotopeDistribution theoretical_iso(EmpiricalFormula(fragment_isotopomer_theoretical_formula).getIsotopeDistribution(CoarseIsotopePatternGenerator(fragment_isotopomer_measured.size())));
+    for (IsotopeDistribution::ConstIterator it = theoretical_iso.begin(); it != theoretical_iso.end(); ++it)
+    {
+      fragment_isotopomer_theoretical.push_back( it->getIntensity() );
+    }
+    
+    std::vector<double> fragment_isotopomer_abs_diff;
+    for (size_t i = 0; i < fragment_isotopomer_theoretical.size(); ++i)
+    {
+      fragment_isotopomer_abs_diff.push_back(std::fabs(fragment_isotopomer_theoretical[i] - fragment_isotopomer_measured[i]));
+    }
+    
+    if (!fragment_isotopomer_abs_diff.empty())
+    {
+      double diff_mean = std::accumulate(fragment_isotopomer_abs_diff.begin(), fragment_isotopomer_abs_diff.end(), 0.0) / fragment_isotopomer_abs_diff.size();
+    
+      diff_mean = OpenMS::Math::MeanAbsoluteDeviation(fragment_isotopomer_abs_diff.begin(), fragment_isotopomer_abs_diff.end(), diff_mean);
+    
+      feature_with_accuracy_info.setMetaValue("average_accuracy", diff_mean);
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateMDVAccuracies(
+    const FeatureMap& normalized_featureMap,
+    FeatureMap& featureMap_with_accuracy_info,
```

Comment:
clear it before adding stuff to it? You never know what is being passed into the function...